crates/flux-infer/src/infer.rs

```diff
@@ -530,7 +542,7 @@ impl Sub {
                 }
                 Ok(())
             }
-            _ => Err(query_bug!("incompatible types: `{a:?}` - `{b:?}`"))?,
+            _ => panic!("incompatible types: `{a:?}` - `{b:?}`"), // _ => Err(query_bug!("incompatible types: `{a:?}` - `{b:?}`"))?,
```

Comment:
Did you do this to get a backtrace? Maybe we can store the stack frame in the `QueryErr::Bug` to get the best of both worlds.

```suggestion
            _ => Err(query_bug!("incompatible types: `{a:?}` - `{b:?}`"))?,
```